### DISCUSSION TITLE
How to enforce first Node to be heading H1 in tiptap editor?

### DISCUSSION THREAD

**Discussion by ghost** [ueberdosis/tiptap#1726] [General Questions] [answered]
Want to achieve something similar to this: [slate.js example](https://www.slatejs.org/examples/forced-layout)

Is this possible using a custom schema?

Tried using a separate <input /> for the title but that breaks things (cursors navigation, backpace, newline).

There is also a Stackoverflow question with no solution: https://stackoverflow.com/questions/67358648/how-to-enforce-first-node-to-be-heading-h1-in-tiptap-editor

**Comment by philippkuehn** [comment, marked as answer]
Absolutely. You can extend the document extension for that.

```js
import { Document } from '@tiptap/extension-documenz'

const CustomDocument = Document.extend({
  content: 'heading block+',
})
```

Read more about the schema [here](https://www.tiptap.dev/api/schema/).

**Reply by ghost** [reply]
Oh, it's that easy. Thanks Philipp!
